sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -544,7 +544,14 @@ case class InSet(child: Expression, hset: Set[Any]) extends UnaryExpression with
 
   require(hset != null, "hset could not be null")
 
-  override def toString: String = s"$child INSET ${hset.mkString("(", ",", ")")}"
+  override def toString: String = {
+    val listString = hset.toSeq
+      .map(elem => Literal(elem, child.dataType).toString)
+      // Sort elements for deterministic behaviours
+      .sorted
```

Comment:
Can we switch the above two lines to support numeric set in a more natural way?
```scala
.sorted
.map(elem => Literal(elem, child.dataType).toString)
```

Reply:
We cannot because `Seq[Any]` cannot be sorted;
```
[error] /Users/maropu/Repositories/spark/spark-master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala:648:8: No implicit Ordering defined for Any.
[error]       .sorted
```

Reply:
Got it. Thank you for checking.